Libraries/LibWeb/Credentials/CredentialsContainer.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2024, Miguel Sacristán <miguel_tete17@hotmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibIDL/Types.h>
+#include <LibWeb/Bindings/CredentialsContainerPrototype.h>
+#include <LibWeb/Bindings/PlatformObject.h>
+#include <LibWeb/DOM/AbortSignal.h>
+#include <LibWeb/WebIDL/Promise.h>
+
+namespace Web::Credentials {
+
+class CredentialsContainer final : public Bindings::PlatformObject {
+    WEB_PLATFORM_OBJECT(CredentialsContainer, Bindings::PlatformObject);
+    GC_DECLARE_ALLOCATOR(CredentialsContainer);
+
+public:
+    static WebIDL::ExceptionOr<GC::Ref<CredentialsContainer>> construct_impl(JS::Realm&);
+
+    [[nodiscard]] GC::Ref<WebIDL::Promise> get(CredentialRequestOptions const&);
+
+private:
+    explicit CredentialsContainer(JS::Realm& realm);
+
+    void initialize(JS::Realm& realm) override;
```

Comment:
Make sure to add `virtual` to the front of overridden virtual methods, per the style guide